src/annotator/integrations/pdf.js

```diff
@@ -241,13 +261,26 @@ export class PDFIntegration {
       return;
     }
 
-    this._warningBanner = document.createElement('hypothesis-banner');
-    document.body.prepend(this._warningBanner);
+    if (!this._banner) {
+      this._banner = document.createElement('hypothesis-banner');
+      document.body.prepend(this._banner);
+      createShadowRoot(this._banner);
+    }
 
-    const warningBannerContent = createShadowRoot(this._warningBanner);
-    render(<WarningBanner />, warningBannerContent);
+    render(
+      <>
+        {this._bannerState.contentPartner && (
+          <ContentPartnerBanner
+            provider={this._bannerState.contentPartner}
+            onClose={() => this._updateBannerState({ contentPartner: null })}
+          />
+        )}
+        {this._bannerState.noTextWarning && <WarningBanner />}
+      </>,
+      /** @type {ShadowRoot} */ (this._banner.shadowRoot)
+    );
```

Comment:
If we're potentially rendering two banners, we may need to render an outer container for them. I'll think on that.